hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieMergeHandle.java

```diff
@@ -101,33 +103,44 @@
   protected long updatedRecordsWritten = 0;
   protected long insertRecordsWritten = 0;
   protected boolean useWriterSchema;
+  protected boolean populateMetaFields;
+  protected Option<BaseKeyGenerator> keyGeneratorOpt;
```

Comment:
I initially thought of moving this to HoodieWriteHandle thinking all handles might need this. but looks like neither create nor append handle needs key gen since on the write path, we have HoodieKey handy.